numpy/core/_methods.py

```diff
@@ -173,12 +184,13 @@ def _mean(a, axis=None, dtype=None, out=None, keepdims=False):
 
     return ret
 
-def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False, *,
+         where=True):
     arr = asanyarray(a)
 
-    rcount = _count_reduce_items(arr, axis)
+    rcount = _count_reduce_items(arr, axis, keepdims=keepdims, where=where)
```

Comment:
Can you just pass `keepdims=True` here to save needing the reshape below?

Reply:
I think not, let me try to explain:
- `arrmean` in line 204 has to be calculated with `keepdims=True`, otherwise the substraction from the array can be faulty.
- If the user specifies `keepdims=False` for `var`, `rcount` is thus calculated with it. The broadcasting of `arrmean` e.g. shape `(N, 1)` and `rcount` e.g. shape `(N, )` in line 208 at `true_divide` leads to an array of shape `(N, N)` that can no longer be stored in `out`.
- This means that the division in line 208 should definitely happen with a _shaped_ `rcount`.
- However if we would set `keepdims=True` in line 191 at the calculation of `rcount`, we would not be able to use `true_divide` with `ret` and `rcount` in line 238. Let's say `ret` is of shape `(N, )` because the user wants `keepdims=False` and `rcount` would be of shape `(N, 1)`, the result would be an array of shape `(N, N)` which could not be stored in the `out` target.

Yesterday, I was going back and forth with this 'dependency tree'. The proposed solution seemed the least intrusive and most efficient to me. However if you see a better solution given the above mentioned limitations, I am happy to try it out :smiley: